operator/v1/types_ingress.go

```diff
@@ -382,6 +389,130 @@ const (
 	StrictNamespaceOwnershipCheck NamespaceOwnershipCheck = "Strict"
 )
 
+// LoggingDestinationType is a type of destination to which to send log
+// messages.
+//
+// +kubebuilder:validation:Enum=Container;Syslog
+type LoggingDestinationType string
+
+const (
+	// Container sends log messages to a sidecar container.
+	ContainerLoggingDestinationType LoggingDestinationType = "Container"
+
+	// Syslog sends log messages to a syslog endpoint.
+	SyslogLoggingDestinationType LoggingDestinationType = "Syslog"
+
+	// ContainerLoggingSidecarContainerName is the name of the container
+	// with the log output in an ingress controller pod when container
+	// logging is used.
+	ContainerLoggingSidecarContainerName = "logs"
+)
+
+// SyslogLoggingDestinationParameters describes parameters for the Syslog
+// logging destination type.
+type SyslogLoggingDestinationParameters struct {
+	// address is the IP address of the syslog endpoint that receives log
+	// messages.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Address string `json:"address"`
```

Comment:
@sttts @deads2k any guidance on the best way to represent and validate endpoints (addr/port, ipv4/ipv6)? Are there any meaningful static validations we can apply in openapi?

Reply:
https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types_jsonschema.go#L53

There is an URL. Otherwise, we split host and port usually, don't we?

Reply:
By "split host and port", do you mean using separate fields?  I changed to using separate fields to enable us to perform stricter validation.  Our goal is to perform as much validation as possible using OpenAPI.